_drafts/zephyr-ring-buffers.md

```diff
@@ -0,0 +1,329 @@
+---
+title: "Zephyr Deep Dive: Ring Buffers"
+description: "Overview of Zephyr ring buffers covering how they work and when to use them in your design"
+author: ejohnso49
+---
+
+Zephyr includes many built-in features like stacks for networking and BLE, Flash
+storage APIs, and many kernel services. These components allow you to quickly
+get up and running with a project and maintain less code! Taking advantage of
+these is a huge win for small firmware teams. This was a huge motivation when I
+first learned about the project and brought it to my teams. Starting to work
+with such a full-featured project is not without its difficulties.
+
+<!-- excerpt start -->
+
+This post will cover Zephyr's built-in ring buffer API. This component is
+commonly used in producer-consumer scenarios. We will cover how the ring buffers
+in Zephyr work, when to use them, and their strengths and weaknesses. This post
+will close with an example of augmenting ring buffers with waiting capabilities.
+
+<!-- excerpt end -->
+
+{% include newsletter.html %}
+
+{% include toc.html %}
+
+## Zephyr Ring Buffers
+
+Ring buffers[^1] in Zephyr provide a way to pass data through a shared memory
+buffer. Ring buffers are safely used in single-consumer, single-producer uses.
+They offer two interfaces, a low-level bytes-based interface, and an items-based
+interface. The items API associates an application-defined type and integer
+value with each write to the ring buffer.
+
+### Bytes API
+
+We will now walk through code from the `es-wifi` driver[^2] that uses a ring
```

Comment:
Maybe add an intro paragraph, like:

```
The Zephyr ring buffer Bytes API is often used in driver implementations, where a piece of hardware needs to send chunks of data up to a higher layer for further processing.
```

or something similar to set up this section

Reply:
Done!